apps/hash-api/src/graph/ensure-system-graph-is-initialized/migrate-ontology-types/migrations/006-create-first-custom-data-types.migration.ts

```diff
@@ -25,48 +28,62 @@ const migrate: MigrationFunction = async ({
       format: "email",
       type: "string",
     },
+    conversions: {},
     webShortname: "hash",
     migrationState,
   });
 
   await createSystemDataTypeIfNotExists(context, authentication, {
     dataTypeDefinition: {
-      title: "Millimeters",
+      title: "Meters",
       description:
-        "A unit of length in the International System of Units (SI), equal to one thousandth of a meter.",
+        "The base unit of length in the International System of Units (SI).",
       label: {
-        right: "mm",
+        right: "m",
       },
       type: "number",
     },
+    conversions: {},
     webShortname: "hash",
     migrationState,
   });
 
   await createSystemDataTypeIfNotExists(context, authentication, {
     dataTypeDefinition: {
-      title: "Centimeters",
+      title: "Millimeters",
       description:
-        "A unit of length in the International System of Units (SI), equal to one hundredth of a meter.",
+        "A unit of length in the International System of Units (SI), equal to one thousandth of a meter.",
       label: {
-        right: "cm",
+        right: "mm",
       },
       type: "number",
     },
+    conversions: {
+      [systemDataTypes.meters.dataTypeBaseUrl]: {
+        from: { expression: ["*", "self", { const: 1000, type: "number" }] },
```

Comment:
I think this would be more easily understandable and future proof if the expression was in a regular order of operations, which meant that you could also add more operators and values on the end. Also figuring out how we have sub-expressions.

E.g. temperature conversion is a good example. It has multiple steps and a subexpression
```
// °F into °C
[["self", "-", { const: "32", type: "number" }] "*", { const: 5/9, type: "number" }]
```

Reply:
Alternative question – how is the above expressed in the intended language in the other PR?

Reply:
The format is similar to S-Expressions in prefix notation but in JSON syntax for easier implementation. The first element always determines the function/operator, subsequent elements represents arguments which can be an expression again. Two examples + to alternatives were given in the test-code for conversions:
```typescript
// °F into °C, "(self - 32) * 5 / 9"
[
    "/",
    [
        "*",
        [
            "-",
            "self",
            { "const": 32.0, "type": "number" }
        ],
        { "const": 5.0, "type": "number" }
    ],
    { "const": 9.0, "type": "number" },
]
```
```typescript
// °F into °C alternative, "(self - 32) * (5 / 9)"
[
    "*",
    [
        "-",
        "self",
        { "const": 32.0, "type": "number" }
    ],
    [
        "/",
        { "const": 5.0, "type": "number" },
        { "const": 9.0, "type": "number" }
    ]
]
```
```typescript
// °C into °F, "self * 9 / 5 + 32"
[
    "+",
    [
        "/",
        [
            "*",
            "self",
            { "const": 9.0, "type": "number" }
        ],
        { "const": 5.0, "type": "number" }
    ],
    { "const": 32.0, "type": "number" },
]
```
```typescript
// °C into °F alternative, "self * (9 / 5) + 32"
[
    "+",
    [
        "*",
        "self",
        [
            "/",
            { "const": 9.0, "type": "number" },
            { "const": 5.0, "type": "number" }
        ]
    ],
    { "const": 32.0, "type": "number" },
]
```

There is also an implementation provided to make this readable (e.g. `"self * (9 / 5) + 32"`).

The types are pretty error-proof:
```ts
type ConversionValue = 
  | "self"
  | { const: number; type: "number" }
  | ["+" | "-" | "*" | "/", ConversionValue, ConversionValue];
```

Reply:
Ahh, sorry, I missed that file! Makes sense, thank you